File: components/si-web-app/src/api/sdf/model/siStorable.ts
import { IUpdateClock } from "./updateClock";

export interface ISiStorable {
  typeName: string;
  objectId: string;
  billingAccountId: string;
  organizationId: string;
  workspaceId: string;
  tenantIds: string[];
  createdByUserId?: string;
  updateClock: IUpdateClock;
  deleted: boolean;
}

export interface ISimpleStorable {
  typeName: String;
  objectId: String;
  billingAccountId: String;
  tenantIds: String[];
  deleted: boolean;
}

export interface IMinimalStorable {
  typeName: String;
  objectId: String;
  deleted: boolean;
}

File: components/si-model/src/si_storable.rs
use chrono::prelude::{DateTime, Utc};
use serde::{Deserialize, Serialize};

#[derive(Serialize, Deserialize, Debug, PartialEq, Eq, Clone)]
#[serde(rename_all = "camelCase")]
pub struct SiStorable {
    pub type_name: String,
    pub object_id: String,
    pub billing_account_id: String,
    pub organization_id: String,
    pub workspace_id: String,
    pub tenant_ids: Vec<String>,
    pub created_at: Option<DateTime<Utc>>,
    pub updated_at: Option<DateTime<Utc>>,
    pub deleted: bool,
}

#[derive(Deserialize, Serialize, Debug, Clone, Eq, PartialEq)]
#[serde(rename_all = "camelCase")]
pub struct SimpleStorable {
    pub type_name: String,
    pub object_id: String,
    pub billing_account_id: String,
    pub tenant_ids: Vec<String>,
    pub deleted: bool,
}

#[derive(Deserialize, Serialize, Debug, Clone, Eq, PartialEq)]
#[serde(rename_all = "camelCase")]
pub struct MinimalStorable {
    pub type_name: String,
    pub object_id: String,
    pub deleted: bool,
}
